pkg/network/render.go

```diff
@@ -318,3 +326,76 @@ func RenderMultus(conf *operv1.NetworkSpec, manifestDir string) ([]*uns.Unstruct
 	out = append(out, objs...)
 	return out, nil
 }
+
+// RenderDHCP generates the manifests for the reference DHCP CNI plugin running as a daemon
+func RenderDHCP(conf *operv1.NetworkSpec, manifestDir string) ([]*uns.Unstructured, error) {
+
+	// This isn't useful with Multinetwork.
+	if *conf.DisableMultiNetwork {
+		return nil, nil
+	}
+
+	renderdhcp := false
+
+	// Look and see if we have an AdditionalNetworks
+	if conf.AdditionalNetworks != nil {
+
+		for _, addnet := range conf.AdditionalNetworks {
+
+			// Parse the RawCNIConfig
+			var rawConfig map[string]interface{}
+			var err error
+
+			confBytes := []byte(addnet.RawCNIConfig)
+			err = json.Unmarshal(confBytes, &rawConfig)
+			if err != nil {
+				log.Printf("WARNING: Not rendering DHCP daemonset, failed to Unmarshal RawCNIConfig: %v", confBytes)
+				return nil, nil
+			}
+
+			// Cycle through the IPAM keys, and determine if the type is dhcp
+			if rawConfig["ipam"] != nil {
+				ipam := rawConfig["ipam"].(map[string]interface{})
```

Comment:
This will panic if the CNI is invalid (`"ipam": "yes"`). Need to do a checked typecast.

Likewise below. Alternatively, demarshal it into a real type and make `json.Unmarshal` do the checking for you:

    type CNINetConfIPAMOnly struct {
    	IPAM struct {
    		Type string `json:"type"`
    	} `json:"ipam"`
    }